### PR TITLE
Allow repo.yaml to specify a subdir for applications

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge allows the repo.yaml in an application repository to specify they subdirectory the applications should live in.

The default is `applications`. Empty strings can be passed in to allow no subdirectory.

This replicates the functionality from a PR in spack https://github.com/spack/spack/pull/36643

### CODE REVIEW

File: lib/ramble/ramble/test/repository.py [hunk 1]
```diff
@@ -12,18 +12,20 @@
 import ramble.paths
 
 
-@pytest.fixture()
-def extra_repo(tmpdir_factory):
+@pytest.fixture(params=["applications", "", "foo"])
```
Comment (rfbgo) [inline lib/ramble/ramble/test/repository.py:15]: Is it easy to add an end_to_end style test too? I don't love that this adds a user exposed API/CLI surface (`-d`) and doesn't explicitly use it

File: lib/ramble/ramble/repository.py [hunk 2]
```diff
@@ -1052,7 +1055,7 @@ def __contains__(self, app_name):
         return self.exists(app_name)
 
 
-def create_repo(root, namespace=None):
+def create_repo(root, namespace=None, subdir=applications_dir_name):
```
Comment (rfbgo) [inline lib/ramble/ramble/repository.py:1058]: Should we make `applications_dir_name` more obviously `const` (ie enum member or similar) if we're making such that a user changing the dir does not update the var?

**Review by rfbgo — CHANGES_REQUESTED** [review]
LGTM, 2 questions

File: lib/ramble/ramble/test/repository.py (hunk 1 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/test/repository.py:15]: This should be fixed. I added some tests in `cmd/repo.py` to actually use this feature.

**Review by rfbgo — APPROVED** [review]
Thanks for the changes, this is really nice now

**Pull request merged**